biz.aQute.bndlib/src/aQute/bnd/component/AnnotationReader.java

```diff
@@ -601,20 +721,30 @@ protected void doComponent(Component comp, Annotation annotation) throws Excepti
 				component.service[i] = ref;
 			}
 		}
+		
+		//make sure reference processing knows this is a Reference in Component
+		member = null;
 		Object[] refAnnotations = annotation.get("reference");
 		if (refAnnotations != null) {
 			for (Object o: refAnnotations) {
 				Annotation refAnnotation = (Annotation)o;
-				LookupReference ref = refAnnotation.getAnnotation();
-				ReferenceDef refdef = new ReferenceDef();
-				refdef.name = ref.name();
-				refdef.service = analyzer.getTypeRef((String) refAnnotation.get("service")).getFQN();
-				refdef.cardinality = ref.cardinality();
-				refdef.policy = ref.policy();
-				refdef.policyOption = ref.policyOption();
-				refdef.target = ref.target();
-				refdef.scope = ref.scope();
-				component.references.put(refdef.name, refdef);
+				Reference ref = refAnnotation.getAnnotation();
+				doReference(ref, refAnnotation);
+//				ReferenceDef refdef = new ReferenceDef();
+//				refdef.name = ref.name();
+//				refdef.bind = ref.bind();
+//				refdef.updated = ref.updated();
+//				refdef.unbind = ref.unbind();
+//				refdef.field = ref.field();
+//				refdef.fieldOption = ref.fieldOption();
+//				
+//				refdef.service = analyzer.getTypeRef((String) refAnnotation.get("service")).getFQN();
+//				refdef.cardinality = ref.cardinality();
+//				refdef.policy = ref.policy();
+//				refdef.policyOption = ref.policyOption();
+//				refdef.target = ref.target();
+//				refdef.scope = ref.scope();
+//				component.references.put(refdef.name, refdef);
```

Comment:
Why is this commented out? Perhaps a comment should indicate when it can be uncommented?